src/packet_router/mod.rs

```diff
@@ -55,26 +56,38 @@ pub struct PacketRouter {
     transmit: gateway::MessageSender,
     service: PacketRouterService,
     reconnect: Reconnect,
+    ack_timer: AckTimer,
     store: MessageCache<PacketUp>,
 }
 
+impl MessageHash for PacketUp {
+    fn hash(&self) -> Vec<u8> {
+        Sha256::digest(&self.payload).to_vec()
+    }
+}
+
 impl PacketRouter {
     pub fn new(
         settings: &Settings,
         messages: MessageReceiver,
         transmit: gateway::MessageSender,
     ) -> Self {
         let router_settings = &settings.router;
-        let service =
-            PacketRouterService::new(router_settings.uri.clone(), settings.keypair.clone());
+        let service = PacketRouterService::new(
+            router_settings.uri.clone(),
+            router_settings.ack_timeout(),
+            settings.keypair.clone(),
+        );
         let store = MessageCache::new(router_settings.queue);
         let reconnect = Reconnect::default();
+        let ack_timer = AckTimer::new(router_settings.ack_timeout());
```

Comment:
do we want to add some leeway here? the router currently sends exactly on this interval, so maybe add 1 or 2 seconds?

Reply:
addressed in [697c526](https://github.com/helium/gateway-rs/pull/475/commits/697c5261dfdbbc597123224b0a529b06d7c51031)